docs/enhancements/install-strategy.md

```diff
@@ -0,0 +1,196 @@
+
+### Non-Goals
+
+- Replace ClusterProvision. This CRD represents an attempt to install a standalone cluster, it's highly specific to running openshift-install in a pod, and we do not want to modify anything in this area. Even if we move standalone provisions to an OpenShiftInstallStrategy, ClusterProvision will likely be used beneath it as the retry mechanism.
+- Implementing retries within Hive itself. Retries will be left to specific implementations of InstallStrategy interface, if desired.
+
+## Proposal
+
+Introduce a new InstallStrategy CRD interface/contract in Hive API, breaking out of the current ClusterDeployment.Spec and instead linking to one and use duck typing to interact. Initial implementation would be AgentInstallStrategy. In the future we would hopefully migrate current openshift-install pods to OpenShiftInstallStrategy, and add HypershiftInstallStrategy, etc.
+
+Hive InstallStrategy implementations are a day 1 provisioning API. They would adhere to a contract with hive for what fields/conditions they must contain and update as the installation proceeds. Hive would copy relevant data back to the ClusterDeployment as appropriate at various stages of the provisioning.
+
+ClusterDeployment becomes a central index of clusters and kubeconfigs, used for day 2 management.
+
+AgentInstallStrategy becomes a day 1 resource and will offer more flexibility for the agent controllers to define settings they need in isolation, and track their relevant status.
+
+
+### User Stories
+
+#### Story 1
+
+#### Story 2
+
+## Design Details
+
+
+### ClusterDeployment Changes
+
+ClusterDeployment should now contain only information needed to connect to the cluster and settings for day 2 management. (admin kubeconfig, admin password, ingress, control plane certs, etc) Platform section will remain for information we need, including cloud provider credentials for things like hibernation and managed DNS.
+
+ClusterDeployment.Spec.InstallStrategy goes away, and is instead replaced with InstallStrategyRef, which would look like the following:
+
+```yaml
+  installStrategyRef:
+    apiVersion: hive.openshift.io/v1
+    kind: AgentInstallStrategy
+    name: mycluster
+```
+
+InstallStrategyRef is optional, it would not be specified for an adopted cluster.
+
+Hive controllers would then interact with this strategy dynamically with duck typing, similar to Cluster API, Knative, etc.
+
+ClusterDeployment.Spec.Provisioning remains but will eventually be deprecated. Validate this is not used in conjunction with ClusterDeployment.Spec.InstallStrategyRef, must choose one path or the other.
+
+### InstallStrategy Interface
+
+
+Field | Type | Description
+----- | ---- | -----------
+spec.clusterDeploymentRef | LocalObjectReference | Link to the required ClusterDeployment for this install. Bidirectional reference. Common data should live on ClusterDeployment and never be duplicated on InstallStrategy.
```

Comment:
so if i understand the purpose of  this field in the ClusterInstall is to support fetching information that is common?

can provide examples of the class of information that would be on ClusterDeployment that can be use by these ClusterInstall implementations.

Reply:
I wonder if an owner reference would suffice.

Reply:
Examples are primarily things in CD.Spec.Platform.[CloudProvider] which I think the installers will often need and we would not want to duplicate. Felt like it made sense to not-duplicate anything, and assume that an Install must link up with a ClusterDeployment.